doc/layer2-forwarding-enhancements/SONiC Layer 2 Forwarding Enhancements HLD.md

```diff
@@ -0,0 +1,444 @@
+
+
+When Spanning tree state changes, protocol component within orchagent will call fdborch API to flush FDB by either per Port or per Port per VLAN. Only dynamic FDB will be flushed.
+
+
+	**FDB Flush due to Port operational state down:**
+
+
+<img src="images/portDownFlush.jpg" width="800" height="500">
+
+
+	**FDB Flush due to Port removal from VLAN:**
+
+
+<img src="images/portVlanRemoveFlush.jpg" width="800" height="500">
+
+
+	**FDB Flush due to spanning tree state change:**
+
+
+<img src="images/stpFlush.jpg" width="800" height="500">
+
+
+**Changes for handling MAC move events.**
+Hardware can generate a single MAC move event instead of generating 2 events (del-old mac and add-new mac). Orchagent code changes will be done to replace the existing FDB entry with the new port in STATE_DB. 
+
+
+**Changes for FDB aging time configuration.**
+A new CLI commands will be added for configuring the FDB aging time.
+FDB aging time configuration will be set in the SWITCH table in CONFIG_DB. Vlanmgr will populate it in the APP_DB. Orchagent will get notified and handling will be SwitchOrch. SwitchOrch will call sai_redis API to send request to syncd via ASIC_DB.
+
+
+<img src="images/agingTimeConfig.jpg" width="800" height="500">
+
+
+**Changes for Static FDB configuration.**
+New CLI will be added for configuring Static FDB entry. It will be set in the FDB_TABLE in CONFIG_DB. Vlanmgr will handle the CONIG_DB changes, do necessary validations and then populate in APP_DB. FdbOrch already has all the necessary handling for Static FDB changes from APP_DB.
+When a port is removed from VLAN, the corresponding Static FDB entries will also be flushed from all databases except APP_DB and CONFIG_DB.  Flushed static FDB will be stored in the saved FDB in orchagent for retrieval and programming later when port is added back to VLAN.
+If a dynamic FDB is already learnt and a static FDB is configured with same (MAC, VLAN), the existing dynamic FDB entry will be replaced with the static entry.
+
+
+
+
+<img src="images/staticMacConfig.jpg" width="800" height="500">
+
+
+
+
+### 3.3.2 Other Process 
+**Vlanmgr changes**
+
+Vlanmgr will handle Aging time configuration changes from SWITCH table in CONFIG_DB. It will set the new aging time in SWITCH_TABLE in APP_DB, which will be processed by SwitchOrch.
+
+
+Vlanmgr will handle Static FDB configuration from FDB table in CONFIG_DB. It will do validation and then set the static FDB in FDB_TABLE in APP_DB, which will be processed by FdbOrch.
+
+
+## 3.4 SyncD
+No change.
+
+
+## 3.5 SAI
+No changes are being made in SAI. The following SAI attributes are used by the changes being made to SONiC:
+
+1. SAI_FDB_FLUSH_ATTR_BRIDGE_PORT_ID
+2. SAI_FDB_FLUSH_ATTR_BV_ID
+3. SAI_FDB_FLUSH_ATTR_ENTRY_TYPE
+4. SAI_FDB_ENTRY_ATTR_TYPE
+5. SA_FDB_EVENT_MOVE
+6. SAI_SWITCH_ATTR_FDB_AGING_TIME
+
+
+
+## 3.6 CLI
+
+### 3.6.1 Configuration Commands
+**FDB Aging time configuration**
+
+root@sonic:/# config mac aging-time <value>
```

Comment:
the current FDB flush command "sonic-clear fdb xxx", do you have the plan to align the convention?  Either both use "fdb" or "mac"?

Reply:
The config commands will use 'mac'. Existing clear commands are unchanged.